File: android/app/src/main/java/com/example/conceirge/data/models/LibraryItem.kt
package com.example.conceirge.data.models

data class LibraryItem(
    val id: Int,
    val tmdb_id: Int,
    val title: String,
    val year: Int,
    val overview: String?,
    val poster_url: String?,
    val status: String, // "searching"|"grabbing"|"downloading"|"downloaded"|"in_library"|"failed"
    val fail_reason: String?,
    val chosen_release_title: String?,
    val chosen_release_size_gb: Double?,
    val chosen_release_quality: String?,
    val download_id: String?,
    val download_progress: Double?, // 0–100, fetched live from SABnzbd; null when not downloading
    val radarr_has_file: Boolean = false,
    val created_at: String,
    val updated_at: String
)

File: backend/models/library_item.py
from datetime import datetime, timezone
from typing import Literal

from sqlmodel import Field, SQLModel

LibraryStatus = Literal[
    "searching", "grabbing", "downloading", "downloaded", "in_library", "failed"
]


class LibraryItem(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    tmdb_id: int = Field(index=True)
    radarr_movie_id: int | None = Field(default=None)
    title: str
    year: int
    overview: str | None = None
    poster_url: str | None = None
    status: str = "searching"
    fail_reason: str | None = None
    chosen_release_title: str | None = None
    chosen_release_size_gb: float | None = None
    chosen_release_quality: str | None = None
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
